pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -313,3 +313,5 @@ func dropDeallocatedStatusDevices(newClaim, oldClaim *resource.ResourceClaim) {
 		newClaim.Status.Devices = nil
 	}
 }
+
+// TODO: add tests after partitionable devices is merged (code conflict!)
```

Comment:
@pohly should we sequence this PR after partitionable merges? There are also changes in partitionable that add a `features` struct for the feature gates in allocator.

Reply:
@cici37: you can remove this comment as part of your rebase. I already added those tests, as you will see while resolving conflicts.

Reply:
I copy-and-pasted as much of your test changes as possible, you should be able to simple add your new test cases.

Reply:
Thanks. Will review while rebasing^^